### PR TITLE
implement defVars macro

### PR DESCRIPTION
**Pull request by mlubin**
Now with parentheses syntax, see the tests. The parsing *should* work fine for valid expressions, but we will probably need to add a bit more to catch syntax errors.

Closes https://github.com/JuliaOpt/JuMP.jl/pull/445
Closes https://github.com/JuliaOpt/JuMP.jl/issues/440

### CODE REVIEW

File: test/macros.jl [hunk 1]
```diff
@@ -452,3 +452,31 @@ facts("[macros] Issue #621") do
...
     @fact conToStr(con) --> "3 x² + x + 1 $leq 0"
 end
+
+facts("[macros] @defVars and @addConstraints") do
+    m = Model()
+    @defVars m begin
+        0 ≤ x ≤ 1
+        y ≥ 2, Int
+        z ≤ 3, (start=10)
+        q, (Bin, start=0.5)
```
Comment (joehuchette) [inline test/macros.jl]: Is
```jl
q, Bin, (start=0.5)
```
valid as well here?

Comment (mlubin) [inline test/macros.jl]: Yes

**Comment by mlubin** [comment]
@IainNZ, any objections?

File: test/macros.jl [hunk 2]
```diff
@@ -452,3 +452,32 @@ facts("[macros] Issue #621") do
     con = @QuadConstraint(a+q*3 <= 0)
     @fact conToStr(con) --> "3 x² + x + 1 $leq 0"
 end
+
+facts("[macros] @defVars and @addConstraints") do
+    m = Model()
+    @defVars m begin
+        0 ≤ x ≤ 1
```
Comment (IainNZ) [inline test/macros.jl]: Maybe have a variable with indicies and nontrivial bounds, just to be safe with parsing?

**Comment by IainNZ** [comment]
No objections on technical merit, bit nervous about having yet another syntax

**Comment by mlubin** [comment]
This is the future™

File: test/macros.jl (hunk 2 above)
Comment (mlubin) [inline test/macros.jl]: Updated

**Comment by IainNZ** [comment]
Hah ok. Just had a longish conversation with a couple of the first years who were scared that JuMP was doing weird scoping things, not sure this eases those anxieties. But YOLO.

**Comment by joehuchette** [comment]
What were the scoping concerns?

**Comment by IainNZ** [comment]
Just a lack of confidence that a `@defVar`'d variable was a just a plain-old Julia variable, and the notion of ownership between a variable and a model. Also a desire for `getVar`, because they have 20 or so different sets of variables in their model and were struggling to break down their model construction in a way they felt happy with.

**Comment by mlubin** [comment]
I've been toying with an 
```
x = ....
```
syntax within macros but haven't been able to come up with something reasonable.

**Pull request merged**